.github/workflows/main.yaml

```diff
@@ -93,13 +97,20 @@ jobs:
       fail-fast: false
       matrix:
         python-version: [3.8, 3.9]
+        dependencies: ["releases-only", "upstream-dev"]
         pytest-mark: ["no_executor", "executor_function", "executor_generator",
                       "executor_dask", "executor_prefect", "executor_prefect_dask",
                       "executor_beam"]
         exclude:
           # No beam support in python 3.9
           - python-version: 3.9
             pytest-mark: "executor_beam"
+          - dependencies: "upstream-dev"
+            pytest-mark: "no_executor"
```

Comment:
I would remove this exclusion. Running the `no_executor` tests is pretty cheap and covers lots of core functionality not covered by the executor tests.

```suggestion
```